src/ray/core_worker/core_worker.h

```diff
@@ -586,7 +587,7 @@ class CoreWorker {
   /// A map from resource name to the resource IDs that are currently reserved
   /// for this worker. Each pair consists of the resource ID and the fraction
   /// of that resource allocated for this worker.
-  ResourceMappingType resource_ids_;
+  std::shared_ptr<ResourceMappingType> resource_ids_;
```

Comment:
This should already be there, but add a comment saying this is set/reset on task assignment?